core/deployment/src/main/java/io/quarkus/deployment/dev/DevModeContext.java

```diff
@@ -312,6 +312,13 @@ public CompilationUnit getMain() {
             return Optional.ofNullable(test);
         }
 
+        public void addPrecedingSourcePath(String path) {
```

Comment:
Perhaps `addFirst` would sound more usual?

Reply:
I mean `addFirstSourcePath`

Reply:
actually, maybe just adding to sourcePaths on the beginning will work. I'll try to simplify it